app/graphql/resolvers/tag_report_resolver.rb

```diff
@@ -0,0 +1,56 @@
+module TagReportResolver
+  # Note: These queries have been taken from official Dataclips we've been using since 2019.
+  # TODO: Refactor into Active Record Queries to make it testable.
+  class << self
+    def individual(_object, args, _context)
+      office_id = args[:office_id]
+      # We use UTC as the reports should be consistent for all admins regardless of time zones.
+      sql = "
+      SELECT \"id\", \"name\", \"sum\" from tags INNER JOIN
+        (SELECT SUM(duration), tag_id
+        from individual_events INNER JOIN individual_event_tags ON individual_events.id=individual_event_tags.individual_event_id
+        WHERE \"individual_events\".\"deleted_at\" IS NULL
+        #{office_id == 'all' ? '' : "AND individual_events.office_id=#{office_id}"}
+        AND \"individual_events\".\"status\" = 1
+        AND (date > '#{Time.at(args[:after]).utc}')
+        AND (date < '#{Time.at(args[:before]).utc}')
+        GROUP BY tag_id) as tagDuration
+      ON tagDuration.tag_id=tags.id
+      "
```

Comment:
You could use a heredoc for the query string. You won't have to escape all the quotes then.

```ruby
and_office = office_id == 'all' ? '' : "AND individual_events.office_id=#{office_id}"
sql <<-SQL
  SELECT "id", "name", "sum" from tags INNER JOIN
  (SELECT SUM(duration), tag_id
  from individual_events INNER JOIN individual_event_tags ON individual_events.id=individual_event_tags.individual_event_id
  WHERE "individual_events"."deleted_at" IS NULL
  #{and_office}
  AND "individual_events"."status" = 1
  AND (date > '#{Time.at(args[:after]).utc}')
  AND (date < '#{Time.at(args[:before]).utc}')
  GROUP BY tag_id) as tagDuration
ON tagDuration.tag_id=tags.id
SQL
```

Reply:
nice, didn't know that, thanks!